src/manager.cpp

```diff
@@ -160,13 +179,113 @@ void Manager::deleteSingleFruVpd(
     logging::logMessage(std::string(i_dbusObjPath));
 }
 
+bool Manager::isValidUnexpandedLocationCode(
+    const std::string& i_unexpandedLocationCode)
+{
+    if ((i_unexpandedLocationCode.length() <
```

Comment:
if ((i_unexpandedLocationCode.compare("Ufcs") ==
          constants::STR_CMP_SUCCESS) ||
         (i_unexpandedLocationCode.compare("Umts") ==
          constants::STR_CMP_SUCCESS) )
          {
              return true;
           }
           else
           {
                  return false;
            }
            
  can we directly check whether its Umts or Ufcs ?

Reply:
No.
Length check is required. Consider the input is "Ufc" which is an invalid location code. By checking the length of the location code we can directly return it is an invalid location code without doing string comparison.